colcon_ros/task/ament_python/build.py

```diff
@@ -34,4 +40,58 @@ def __init__(self):  # noqa: D107
             'ament_prefix_path', Path(args.install_base),
             self.context.pkg.name, 'AMENT_PREFIX_PATH', '', mode='prepend')
 
+        # get options from the Python manifest
+        try:
+            env = await get_command_environment(
+                'setup_py', args.build_base, self.context.dependencies)
+        except RuntimeError as e:
+            logger.error(str(e))
+            return 1
+        setup_py_data = get_setup_data(self.context.pkg, env)
+
+        # check if the package index and manifest are being installed
+        data_files = get_data_files_mapping(
+            setup_py_data.get('data_files', []))
+        installs_package_index = False
+        installs_package_manifest = False
+
+        # check if package index and manifest are being installed
+        for source, destination in data_files.items():
+            # work around data files incorrectly defined as not relative
+            if os.path.isabs(source):
```

Comment:
Is this case possible? I get an assertion error if I try to set an absolute path:

    AssertionError: 'data_files' must be relative, '/home/jacob/foo_ws/src/launch/launch/package.xml' is absolute

Maybe we can remove this logic.

Reply:
The same logic is also present in [colcon-core](https://github.com/colcon/colcon-core/blob/bb615dffe8282e5b7364534d90d8e48b06c9b6ea/colcon_core/task/python/build.py#L209). While I agree with you that it looks like this can probably be removed (assuming all code paths already check that assertion) I would keep this defensive code for now and create a set of follow up PRs for removing it.